src/test/scala/org/broadinstitute/dsde/workbench/sam/service/UserServiceSpec.scala

```diff
@@ -0,0 +1,152 @@
+
+  lazy val config = ConfigFactory.load()
+  lazy val directoryConfig = config.as[DirectoryConfig]("directory")
+  lazy val petServiceAccountConfig = config.as[PetServiceAccountConfig]("petServiceAccount")
+  lazy val dirDAO = new JndiDirectoryDAO(directoryConfig)
+
+  var service: UserService = _
+
+  override protected def beforeAll(): Unit = {
+    runAndWait(dirDAO.init())
+  }
+
+  before {
+    service = new UserService(dirDAO, new MockGoogleDirectoryDAO(), new MockGoogleIamDAO(), "dev.test.firecloud.org", petServiceAccountConfig)
+  }
+
+  after {
+    // clean up
+    dirDAO.removePetServiceAccountFromUser(defaultUserId).futureValue
+    dirDAO.deleteUser(defaultUserId).futureValue
+    dirDAO.deleteGroup(service.allUsersGroupName).futureValue
+  }
+
+  "UserService" should "create a user" in {
+    // create a user
+    val newUser = service.createUser(defaultUser).futureValue
+    newUser shouldBe Some(UserStatus(UserStatusDetails(defaultUserId, defaultUserEmail), Map("ldap" -> false, "allUsersGroup" -> true, "google" -> true)))
+
+    // check ldap
+    dirDAO.loadUser(defaultUserId).futureValue shouldBe Some(defaultUser)
+    // TODO: cn=enabled-users isn't created in the test environment, so this check fails
+    //dirDAO.isEnabled(defaultUserId).futureValue shouldBe true
+    dirDAO.loadGroup(service.allUsersGroupName).futureValue shouldBe
+      Some(WorkbenchGroup(service.allUsersGroupName, Set(defaultUserId), WorkbenchGroupEmail(service.toGoogleGroupName(service.allUsersGroupName.value))))
+
+    // check google
+    val mockGoogleDirectoryDAO = service.googleDirectoryDAO.asInstanceOf[MockGoogleDirectoryDAO]
+    val groupEmail = WorkbenchGroupEmail(service.toProxyFromUser(defaultUserId.value))
+    val allUsersGroupEmail = WorkbenchGroupEmail(service.toGoogleGroupName(service.allUsersGroupName.value))
+    mockGoogleDirectoryDAO.groups should contain key (groupEmail)
+    mockGoogleDirectoryDAO.groups(groupEmail) shouldBe Set(defaultUserEmail)
+    mockGoogleDirectoryDAO.groups should contain key (allUsersGroupEmail)
+    mockGoogleDirectoryDAO.groups(allUsersGroupEmail) shouldBe Set(WorkbenchUserEmail(service.toProxyFromUser(defaultUserId.value)))
+  }
+
+  it should "get user status" in {
+    // user doesn't exist yet
+    service.getUserStatus(defaultUserId).futureValue shouldBe None
+
+    // create a user
+    val newUser = service.createUser(defaultUser).futureValue
+    newUser shouldBe Some(UserStatus(UserStatusDetails(defaultUserId, defaultUserEmail), Map("ldap" -> false, "allUsersGroup" -> true, "google" -> true)))
+
+    // user should exist now
+    val status = service.getUserStatus(defaultUserId).futureValue
+    status shouldBe Some(UserStatus(UserStatusDetails(defaultUserId, defaultUserEmail), Map("ldap" -> false, "allUsersGroup" -> true, "google" -> true)))
+  }
+
+  // TODO: cn=enabled-users isn't created in the test environment, so this test fails
```

Comment:
This test fails because the `cn=enabled-users` attribute doesn't exist in the unit test env. It looks like it is not created programmatically in `JndiUserDirectoryDAO`. I didn't want to create it programmatically for risk of screwing up prod, so I just ignored this test.

Reply:
we should definitely figure this out but not part of this PR